make/jdk/src/classes/build/tools/generatecharacter/GenerateCharacter.java

```diff
@@ -776,6 +802,12 @@ static String replaceCommand(String x) {
         if (x.equals("maskIdeographic")) return "0x" + hex4(maskIdeographic >> 32);
         if (x.equals("maskIDStart")) return "0x" + hex4(maskIDStart >> 32);
         if (x.equals("maskIDContinue")) return "0x" + hex4(maskIDContinue >> 32);
+        if (x.equals("maskEmoji")) return "0x" + hex4(maskEmoji >> 32);
+        if (x.equals("maskEmojiPresentation")) return "0x" + hex4(maskEmojiPresentation >> 32);
+        if (x.equals("maskEmojiModifier")) return "0x" + hex4(maskEmojiModifier >> 32);
+        if (x.equals("maskEmojiModifierBase")) return "0x" + hex4(maskEmojiModifierBase >> 32);
+        if (x.equals("maskEmojiComponent")) return "0x" + hex4(maskEmojiComponent >> 32);
+        if (x.equals("maskExtendedPictographic")) return "0x" + hex4(maskExtendedPictographic >> 32);
```

Comment:
An upgrade would be to modify hex4(), hexNN() to use `HexFormat.of().toUpperCase().toHexDigits((short)xxx)`
The HexFormat is reusable and would avoid creating extra strings.
Perhaps also create a method that combines the repetitive shift and prefixing.

This if...then... sequence could be an expression switch (x) {...}.

Reply:
It would be good, but it would be for another day IMO.